File: src/Python/601-700/690.GetImportance.py

# Employee info
class Employee:
    def __init__(self, id, importance, subordinates):
        # It's the unique id of each node.
        # unique id of this employee
        self.id = id
        # the importance value of this employee
        self.importance = importance
        # the id of direct subordinates
        self.subordinates = subordinates
    
class Solution:
    def getemfromid(self, employees, id):
        for em in employees:
            if em.id == id:
                return em
        return None

    def getImportance(self, employees, id):
        """
        :type employees: Employee
        :type id: int
        :rtype: int
        """      
        root = self.getemfromid(employees, id)
        r = root.importance
        for emid in root.subordinates:
            r += self.getImportance(employees, emid)
        return r

if __name__ == '__main__':
    solution = Solution()
    em1 = Employee(1, 5, [2, 3])
    em2 = Employee(2, 3, [])
    em3 = Employee(3, 3, [])
    print(solution.getImportance([em1, em2, em3], 1))
else:
    pass

File: src/CSharp/601-700/690.GetImportance.cs


using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.GetImportance
{

    public class Employee
    {
        // It's the unique id of each node;
        // unique id of this employee
        public int id;
        // the importance value of this employee
        public int importance;
        // the id of direct subordinates
        public List<int> subordinates;
    };

    public class Solution
    {
        int importance = 0;
        public int GetImportance(List<Employee> employees, int id)
        {
            var map = new Dictionary<int, Employee>();
            foreach (var e in employees)
            {
                map.Add(e.id, e);
            }
            Dfs(map[id], map);
            return importance;
        }
        public void Dfs(Employee e, Dictionary<int, Employee> map)
        {
            importance += e.importance;
            foreach (var i in e.subordinates)
            {
                Dfs(map[i], map);
            }
        }
    }
}
